app/src/main/java/com/example/cinemapp/ui/main/ActorDetailsViewModel.kt

```diff
@@ -0,0 +1,79 @@
+package com.example.cinemapp.ui.main
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.cinemapp.data.MovieRepository
+import com.example.cinemapp.ui.main.model.CastMovieCredit
+import com.example.cinemapp.ui.main.model.PersonDetails
+import com.example.cinemapp.util.MovieUtil
+import kotlinx.coroutines.async
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+class ActorDetailsViewModel(
+    private val movieRepository: MovieRepository
+) : ViewModel() {
+
+    data class State(
+        val details: PersonDetails? = null,
+        val creditYears: List<Int?> = emptyList()
+    )
+
+    data class StateCredits(
```

Comment:
`val credits: List<CastMovieCredit> = emptyList()` should be moved to `State` class, thus removing the `StateCredits` class. It will make transition to compose much easier.